src/components/UI/Button/Button.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { FunctionComponent, MouseEventHandler } from 'react';
+import classes from './Button.module.css';
+
+type ButtonProps = {
+  onClick?: MouseEventHandler
+}
```

Comment:
TypeScript hint: you should use `type` for unions or aliases; for generic data shape declaration you should define an `interface`. This might be a bit confusing at first, but interfaces compile slightly quicker, as `tsc` does `type` inference till the end, and in general it could be infinitely recursive.

Reply:
So you can do this with ease at heart:

```ts
type MyApp<A> = ReaderTaskEither<MyAppEnv, Error, A>;
```

but your `ButtonProps` should be:
```ts
interface ButtonProps {
  readonly onClick?: MouseEventHandler;
}
```

Reply:
That's very useful information for me, thank you!
Fixed